cachi2/core/package_managers/pip.py

```diff
@@ -273,74 +273,61 @@ def _generate_purl_dependency(package: dict[str, Any]) -> str:
     return purl.to_string()
 
 
-def _get_pip_metadata(package_dir: RootedPath) -> tuple[str, Optional[str]]:
-    """
-    Attempt to get the name and version of a Pip package.
+def _infer_package_name_from_origin_url(package_dir: RootedPath) -> str:
+    try:
+        repo_id = get_repo_id(package_dir.root)
+    except UnsupportedFeature:
+        raise PackageRejected(
+            reason="Unable to infer package name from origin URL",
+            solution=(
+                "Provide valid metadata in the package files or ensure"
+                "the git repository has an 'origin' remote with a valid URL."
+            ),
+            docs=PIP_METADATA_DOC,
+        )
 
-    First, try to parse the setup.py script (if present) and extract name and version
-    from keyword arguments to the setuptools.setup() call. If either name or version
-    could not be resolved and there is a setup.cfg file, try to fill in the missing
-    values from metadata.name and metadata.version in the .cfg file.
+    repo_name = Path(repo_id.parsed_origin_url.path.removesuffix(".git")).name
+    subpath = package_dir.subpath_from_root
+    resolved_name = Path(repo_name).joinpath(subpath)
+    return canonicalize_name(str(resolved_name).replace("/", "-")).strip("-.")
 
-    :param package_dir: Path to the root directory of a Pip package
-    :return: Tuple of strings (name, version)
-    """
-    name = None
-    version = None
 
+def _extract_metadata_from_config_files(
+    package_dir: RootedPath,
+) -> tuple[Optional[str], Optional[str]]:
     pyproject_toml = PyProjectTOML(package_dir)
     setup_py = SetupPY(package_dir)
     setup_cfg = SetupCFG(package_dir)
 
+    # pyproject.toml
     if pyproject_toml.exists():
-        log.info("Extracting metadata from pyproject.toml")
+        log.debug("Checking pyproject.toml for metadata")
         if pyproject_toml.check_dynamic_version():
-            log.warning("Parsing dynamic metadata from pyproject.toml is not supported")
+            log.warning("Dynamic version parsing from pyproject.toml is not supported")
 
-        name = pyproject_toml.get_name()
-        version = pyproject_toml.get_version()
+        name, version = pyproject_toml.get_name(), pyproject_toml.get_version()
+        if name and version:
+            return name, version
 
-    if None in (name, version) and setup_py.exists():
-        log.info("Filling in missing metadata from setup.py")
-        name = name or setup_py.get_name()
-        version = version or setup_py.get_version()
+    # setup.py
+    if setup_py.exists():
+        log.debug("Checking setup.py for metadata")
+        name, version = setup_py.get_name(), setup_py.get_version()
```

Comment:
Please keep these on separate lines, it's more readable, we don't gain anything with this compound line and we don't lose anything by having it nicely split (all occurrences).